src/managers/config.py

```diff
@@ -83,19 +84,28 @@ class Listener:
 
     Args:
         auth_map: AuthMap representing the auth.protocol and auth.mechanism for the listener
-        scope: scope of the listener, CLIENT, INTERNAL or EXTERNAL
+        scope: scope of the listener, CLIENT, INTERNAL, EXTERNAL or EXTRA
         host: string with the host that will be announced
+        baseport (optional): integer port to offset CLIENT port numbers for EXTRA listeners
         node_port (optional): the node-port for the listener if scope=EXTERNAL
     """
 
     def __init__(
-        self, auth_map: AuthMap, scope: Scope, host: str = "", node_port: int | None = None
+        self,
+        auth_map: AuthMap,
+        scope: Scope,
+        host: str = "",
+        baseport: int = 30000,
+        extra_count: int = -1,
```

Comment:
What is extra_count? This seems obtusely named, maybe could take a more self explaining name?

Reply:
(question) I actually don't understand very well what the extra count does here? If we are using `baseport` and the `auth` schema to offset, why do we need the `extra count`?

Reply:
My mistake was to include `extra_count` in the port calculation. The fact that the `baseport` is unique should be enough.
We DO however need to have unique names for each listener, as they follow the format:
```
<scope>-<auth-scheme>-<security-protocol>
```
So if we have 2 EXTRA listeners, they would both have the same name.

We use the `extra_count` to append to the name:
```
EXTRA_SASL_PLAINTEXT_SCRAM_SHA_512_0  # first provided extra_listener
EXTRA_SASL_PLAINTEXT_SCRAM_SHA_512_1  # second provided extra_listener
```

Reply:
**(niptick)** All right, yes, makes sense. I would maybe just change the default from "-1" to "0" since I'm not really sure it is a good default as it rather sounds an invalid value

Reply:
It is invalid, as we only want to use it when building the listener names when we're using `EXTRA`, the other listeners probably don't need it.